WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostMoreButtonUiStateBuilder.kt

```diff
@@ -42,136 +43,144 @@ class ReaderPostMoreButtonUiStateBuilder @Inject constructor(
+                    iconColor = R.attr.wpColorOnSurfaceMedium,
+                    isSelected = false,
+                    onClicked = onButtonClicked
             )
         }
 
-        menuItems.add(
-                SecondaryAction(
-                        type = REPORT_POST,
-                        label = UiStringRes(R.string.reader_menu_report_post),
-                        labelColor = R.attr.colorOnSurface,
-                        iconRes = R.drawable.ic_block_white_24dp,
-                        iconColor = R.attr.wpColorOnSurfaceMedium,
-                        onClicked = onButtonClicked
-                )
-        )
-        return menuItems
-    }
+    private fun buildShare(onButtonClicked: (Long, Long, ReaderPostCardActionType) -> Unit) =
+        SecondaryAction(
+        type = SHARE,
+        label = UiStringRes(R.string.reader_btn_share),
+        labelColor = R.attr.colorOnSurface,
+        iconRes = R.drawable.ic_share_white_24dp,
+        iconColor = R.attr.wpColorOnSurfaceMedium,
+        onClicked = onButtonClicked
+    )
+
+    private fun buildFollow(isPostFollowed: Boolean, onButtonClicked: (Long, Long, ReaderPostCardActionType) -> Unit) =
+        if (isPostFollowed) {
+            SecondaryAction(
+                    type = FOLLOW,
+                    label = UiStringRes(R.string.reader_btn_unfollow),
+                    labelColor = R.attr.wpColorOnSurfaceMedium,
+                    iconRes = R.drawable.ic_reader_following_white_24dp,
+                    isSelected = true,
+                    onClicked = onButtonClicked
+            )
+        } else {
+            SecondaryAction(
+                    type = FOLLOW,
+                    label = UiStringRes(R.string.reader_btn_follow),
+                    labelColor = R.attr.colorSecondary,
+                    iconRes = R.drawable.ic_reader_follow_white_24dp,
+                    isSelected = false,
+                    onClicked = onButtonClicked
+            )
+        }
+
+    private fun buildBlockSite(onButtonClicked: (Long, Long, ReaderPostCardActionType) -> Unit) =
+            SecondaryAction(
+                    type = BLOCK_SITE,
+                    label = UiStringRes(R.string.reader_menu_block_blog),
+                    labelColor = R.attr.wpColorError,
+                    iconRes = R.drawable.ic_block_white_24dp,
+                    iconColor = R.attr.wpColorError,
+                    onClicked = onButtonClicked
+            )
+
+    private fun buildReportPost(onButtonClicked: (Long, Long, ReaderPostCardActionType) -> Unit) =
+            SecondaryAction(
+                    type = REPORT_POST,
+                    label = UiStringRes(R.string.reader_menu_report_post),
+                    labelColor = R.attr.wpColorError,
+                    iconRes = R.drawable.ic_block_white_24dp,
+                    iconColor = R.attr.wpColorError,
+                    onClicked = onButtonClicked,
```

Comment:
Warning (⚠️): KtLint complains about the trailing comma here and thus the CI fails.

Reply:
Nice piece of info @ParaskP7, as I wasn't running ktlint locally because I thought detekt covers those. 🤦  In 6149d64

Reply:
I can't count how many times I experienced the same. But, the problem is not you not running `KtLint` or `Detekt` not covering `KtLint`, because both don't really matter in this case. The problem is that we have a `KtLint` version more that two years old, which asserts a previous version of Kotlin and as such is results in false positives.

PS: Decommissioning `KtLint` will help us move forward (see #14012).